drivers/MbedCRC.h

```diff
@@ -408,9 +432,25 @@ class MbedCRC
     /** Constructor init called from all specialized cases of constructor
      *  Note: All construtor common code should be in this function.
      */
-    void mbed_crc_ctor(void) const
+    void mbed_crc_ctor(void)
     {
         MBED_STATIC_ASSERT(width <= 32, "Max 32-bit CRC supported");
+
+        _mode = (_crc_table == NULL) ? TABLE : BITWISE;
```

Comment:
Logic here seems reverted, If _crc_table is NULL then we BITWISE computation should be done. 
https://github.com/ARMmbed/mbed-os/pull/6708/files#diff-a353a07f9612be80e818a21f61ec8409L181****